kart/tabular/ogr_import_source.py

```diff
@@ -316,20 +315,23 @@ def has_geometry(self):
         return bool(self.layer_defn.GetGeomFieldCount())
 
     def _check_primary_key_option(self, primary_key_name):
+        self.use_ogc_fid_as_pk = False
         if primary_key_name is None:
             return None
         if primary_key_name:
             ld = self.layer_defn
 
+            for i in range(ld.GetFieldCount()):
+                field = ld.GetFieldDefn(i)
+                if primary_key_name == field.GetName():
+                    return primary_key_name
+
             if primary_key_name == self.ogrlayer.GetFIDColumn():
                 # OGR automatically turns 'ogc_fid' column in postgres into an FID,
                 # and removes it from the list of fields below.
+                self.use_ogc_fid_as_pk = True
```

Comment:
Probably it shouldn't? We could set the [`PGSQL_OGR_FID` config option](https://gdal.org/drivers/vector/pg.html#configuration-options) to some random UUID to prevent it doing this. Then we could just treat an `ogc_fid` field as a regular field

Reply:
... yeah we could do that. But we're not using this except for shapefiles right now so the comment is out of date and the behaviour for non-shapefiles isn't very important. I think best not to change this code too much unless it fixes a bug or we start using it for another OGR import type.